addons/delivery/models/delivery_carrier.py

```diff
@@ -172,12 +172,26 @@ def _is_available_for_order(self, order):
 
         return True
 
-    def available_carriers(self, partner, order):
-        return self.filtered(lambda c: c._match(partner, order))
+    def available_carriers(self, partner, source):
+        return self.filtered(lambda c: c._match(partner, source))
 
-    def _match(self, partner, order):
+    def _match(self, partner, source):
         self.ensure_one()
-        return self._match_address(partner) and self._match_must_have_tags(order) and self._match_excluded_tags(order) and self._match_weight(order) and self._match_volume(order)
+        return (
+            self._match_address(partner)
+            and self._match_must_have_tags(source)
+            and self._match_excluded_tags(source)
+            and self._match_weight(source)
+            and self._match_volume(source)
+        )
+
+    def _get_source_lines(self, source):
```

Comment:
I would remove _get_source_lines and do the direct margin in the method directly. I think it's a bad practice to always assume that `stock.move` and s̀ale.order.line` share the same field name and that it will remains like this in the future